Libplanet.Explorer/Store/MySQLRichStore.cs

```diff
@@ -325,27 +549,44 @@ public void StoreSignerReferences(TxId txId, long txNonce, Address signer)
             int limit = int.MaxValue)
         {
             using QueryFactory db = OpenDB();
-            var query = db.Query("signer_references").Where("signer", signer.ToByteArray())
+            var query = db.Query(SignerRefDbName).Where("signer", signer.ToString())
                 .Offset(offset)
                 .Limit(limit)
                 .Select("tx_id");
             query = desc ? query.OrderByDesc("tx_nonce") : query.OrderBy("tx_nonce");
             return query.OrderBy("tx_nonce")
-                .Get<byte[]>()
-                .Select(bytes => new TxId(bytes));
+                .Get<string>()
+                .Select(txString => new TxId(ByteUtil.ParseHex(txString)));
         }
 
-        public void StoreUpdatedAddressReferences(
-            TxId txId,
-            long txNonce,
-            Address updatedAddress)
+        public void StoreUpdatedAddressReferences<T>(Transaction<T> tx)
+            where T : IAction, new()
         {
-            Insert("updated_address_references", new Dictionary<string, object>
+            string updatedAddressRefFilePath = Path.GetTempFileName();
+            using StreamWriter updatedAddressRefBulkFile = new StreamWriter(
+                updatedAddressRefFilePath);
+            try
+            {
+                foreach (Address address in tx.UpdatedAddresses)
+                {
+                    updatedAddressRefBulkFile.WriteLine(
+                        $"{address.ToString()}," +
+                        $"{tx.Id.ToString()}," +
+                        $"{tx.Nonce}");
```

Comment:
It seems there is no column. Is it okay? If it depends on the table definition's column order, in my thought, it looks hard to update.

Reply:
@moreal
I agree that it would be a challenge to make updates moving forward. But writing the rows this way and bulk loading it together is much faster than inserting them one by one especially during the preloading stage. I'll put some more thoughts into how to make this easier to update in the future. I'd love any suggestions you might have as well!